src/SAWScript/Proof.hs

```diff
@@ -154,8 +193,75 @@ ppProp opts (Prop tm) = ppTerm opts tm
 -- constructor indicating that it has already been proved.
 data Theorem =
   Theorem
-  { thmProp :: Prop
-  , thmStats :: SolverStats
+  { _thmProp :: Prop
+  , _thmStats :: SolverStats
```

Comment:
Is this field redundant? It seems like you could extract `SolverStats` from the `Evidence` field. Is it just used to cache the value so we don't have to recompute it multiple times?

Reply:
I agree, it feels a bit redundant, and it probably makes sense to remove it at some point.  However, at this stage I wanted to minimize observable changes in behavior, so I retained the SolverStats alongside the new Evidence type.